dora/src/semck.rs

```diff
@@ -93,14 +93,15 @@ pub fn check<'ast>(vm: &mut VM<'ast>) {
     implck::check(vm);
     return_on_error!(vm);
 
-    // define internal functions & methods
-    prelude::internal_functions(vm);
     prelude::known_methods(vm);
 
     // check types of expressions in functions
     typeck::check(vm);
     return_on_error!(vm);
 
+    // define internal functions & methods
+    prelude::internal_functions(vm);
```

Comment:
This is called too early, this prevents `flowck` and `returnck` from happening on such functions. I wouldn't move that function after those two calls but also out of `semck`. I would rather add a function `prelude::install_conditional_intrinsics` that we can invoke after `semck::check` in `driver/start.rs`. This will ensure that we do all the semantic checks in `semck`, also this will make a refactor easier that I plan to do (I want to move the semantic analysis to a separate library - for that the `semck` needs to run without the `VM` but with its own data structures, so that we only create a `VM` with a valid program.).

Reply:
Fixed in second commit (feel free to squash after review if you are happy with it).